python/IAM_POLICY_REQUIRED/IAM_POLICY_REQUIRED.py

```diff
@@ -0,0 +1,403 @@
+        policy['PolicyName'] for policy in client.list_attached_user_policies(UserName=username)['AttachedPolicies']
+    ]
+    # Get the policies from the users groups
+    groups = [
+        group['GroupName'] for group in client.list_groups_for_user(UserName=username)['Groups']
+    ]
+    for group in groups:
+        policies.extend([
+            policy['PolicyName'] for policy in client.list_attached_group_policies(GroupName=group)['AttachedPolicies']
+        ])
+
+    return policies
+
+
+def get_role_policies(role_name, client):
+    # Get the users managed policies
+    return [
+        policy['PolicyName'] for policy in client.list_attached_role_policies(RoleName=role_name)['AttachedPolicies']
+    ]
+
+
+def has_policy_attached(resource_name, resource_type, policy_name, client):
+    if resource_type == 'AWS::IAM::User':
+        return policy_name in get_user_policies(resource_name, client)
+    elif resource_type == 'AWS::IAM::Role':
+        return policy_name in get_role_policies(resource_name, client)
+    else:
+        raise ValueError('Unable to handle resource type {}'.format(resource_type))
+
+
+def evaluate_compliance(event, configuration_item, valid_rule_parameters):
+    """Form the evaluation(s) to be return to Config Rules
+
+    Return either:
+    None -- when no result needs to be displayed
+    a string -- either COMPLIANT, NON_COMPLIANT or NOT_APPLICABLE
+    a dictionary -- the evaluation dictionary, usually built by build_evaluation_from_config_item()
+    a list of dictionary -- a list of evaluation dictionary , usually built by build_evaluation()
+
+    Keyword arguments:
+    event -- the event variable given in the lambda handler
+    configuration_item -- the configurationItem dictionary in the invokingEvent
+    valid_rule_parameters -- the output of the evaluate_parameters() representing validated parameters of the Config Rule
+
+    Advanced Notes:
+    1 -- if a resource is deleted and generate a configuration change with ResourceDeleted status, the Boilerplate code will put a NOT_APPLICABLE on this resource automatically.
+    2 -- if a None or a list of dictionary is returned, the old evaluation(s) which are not returned in the new evaluation list are returned as NOT_APPLICABLE by the Boilerplate code
+    3 -- if None or an empty string, list or dict is returned, the Boilerplate code will put a "shadow" evaluation to feedback that the evaluation took place properly
+    """
+    ignored_roles = valid_rule_parameters["ignoredRoles"]
+    ignored_users = valid_rule_parameters["ignoredUsers"]
+    policy_name = valid_rule_parameters['policyName']
+    client = get_client('iam', event)
+
+    if should_ignore_config_item(configuration_item, ignored_roles, ignored_users):
+        return 'COMPLIANT'
+    elif has_policy_attached(configuration_item['resourceName'], configuration_item['resourceType'], policy_name, client):
+        return 'COMPLIANT'
+    else:
+        return 'NON_COMPLIANT'

```

Comment:
Add annotation with the helper build_evaluation_from_config_item()

Reply:
added